rest/config_test.go

```diff
@@ -2928,3 +2929,68 @@ func makeScopesConfigWithDefault(scopeName string, collections []string) *Scopes
 	scopesConfig := makeScopesConfig(scopeName, collections)
 	return &scopesConfig
 }
+
+// TestNotFoundOnInvalidDatabase:
+//   - Create rest tester with large config polling interval
+//   - Insert a bad dbConfig into the bucket
+//   - Manually fetch and load db from buckets
+//   - Assert that the bad config is tracked as invalid config
+//   - Delete the bad config manually and attempt to correct the db config through create db endpoint
+//   - Assert db is removed form invalid db's and is now a running database on server context
+func TestNotFoundOnInvalidDatabase(t *testing.T) {
+	rt := NewRestTester(t, &RestTesterConfig{
+		CustomTestBucket: base.GetTestBucket(t),
+		PersistentConfig: true,
+		MutateStartupConfig: func(config *StartupConfig) {
+			// configure the interval time to not run
+			config.Bootstrap.ConfigUpdateFrequency = base.NewConfigDuration(100 * time.Second)
+		},
+		DatabaseConfig: nil,
+	})
+	defer rt.Close()
+	realBucketName := rt.CustomTestBucket.GetName()
+
+	// create a new invalid db config and persist to bucket
```

Comment:
I'd be inclined to parameterize this test and have it work in two ways, just to be sure this works.

1. Have the initial creation be invalid, mimicking what happens on a sync gateway startup when it sees an invalid config.
2. Set a valid config first, then have config polling pick up an invalid config. I _think_ this should go through the same code path but I'm not 100% sure. This would be a running sync gateway and then having an restore/xdcr to a bucket.

Reply:
1. I feel like this test already does this? I insert directly to the bucket a bad config and call fetch and load configs to actually load the db. I feel this mimics startup of a node and a config already being there in bucket? 
2. This is covered by the test written for the original PR for this work, TestCorruptDbConfigHandling is an example